configMeta/baseWebpackConfig.js

```diff
@@ -0,0 +1,99 @@
+const lodash = require('lodash');
+
+/**
+ * A base class for dynamically generating webpack config using Silverstripe default settings
+ */
+module.exports = class BaseWebpackConfig {
+  vendorChunk = null;
+  config = {};
+
+  /**
+   * Get the webpack config
+   */
+  getConfig() {
+    if (!this.config.entry || lodash.isEmpty(this.config.entry)) {
+      throw new Error('At least one valid entry is required');
+    }
+    return this.config;
+  }
+
+  /**
+   * Split vendor modules out as their own file
+   * @param {string} vendorChunk - name of the vendor file (without extension)
+   * @param {string} regex - regex to match modules against to include them in the vendor file
+   */
+  splitVendor(vendorChunk = 'vendor', regex = /[\\/]node_modules[\\/]/) {
+    if (this.vendorChunk) {
+      throw new Error('Vendor can only be split once. For additional cacheGroups use merge().');
+    }
+
+    this.vendorChunk = vendorChunk;
+
+    this.config.optimization = {
+      splitChunks: {
+        cacheGroups: {
+          vendor: {
+            name: vendorChunk,
+            test: regex,
+            reuseExistingChunk: true,
+            enforce: true,
+            chunks: 'all',
+          }
+        }
+      }
+    };
+
+    // ensure entries are configured to "depend on" the vendor entry if necessary
+    this.#splitVendorFromEntries();
+    return this;
+  }
+
+  /**
+   * The entry object for what needs to be transpiled
+   * @see https://webpack.js.org/concepts/entry-points/
+   * @param {object} entry
+   */
+  setEntry(entry) {
+    if (!entry || lodash.isEmpty(entry)) {
+      throw new Error('entry object must contain at least one valid entry');
+    }
+    this.config.entry = entry;
+    this.#splitVendorFromEntries();
+    return this;
+  }
+
+  /**
+   * If we are splitting a vendor bundle, make sure all entries "depend on" it as appopriate.
+   */
+  #splitVendorFromEntries() {
```

Comment:
The `#` is [private method syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)